### DISCUSSION TITLE
Switch between two commanders.

### DISCUSSION THREAD

**Discussion by 7enHom** [bitcraze/discussions#1155] [Q&A]
Hi!

Here I go again.
I'm having some problems switching between two commands. For example, I want to use send_zdistance_setpoint to set the height, and then use send_setpoint to set the thrust from that altitude. The problem is that when I finish the first command and switch to the next, the crazyflie drops and then taking off again for the next command, it looks like a discontinuous process. And my question is, is there a way to make this process continuous?

Thanks a Ton!

**Comment by knmcguire** [comment]
Hmmm that is a bit strange. Since both the send_zdistance_setpoint and send_setpoint are set on the same commander priority, I wouldn't think there is a difference. 

Just perhaps a step back, what are you planning to do? Do you want to have the crazyflie control its height without using any form of height measurement light with the flowdeck? You can perhaps look at this configuration: [CONFIG_CONTROLLER_PID_IMPROVED_BARO_Z_HOLD](https://github.com/bitcraze/crazyflie-firmware/blob/master/src/modules/src/controller/position_controller_pid.c). If you build the firmware with this enabled in kbuild, then if you use the controller and enable height hold (through assist mode in the client) after takeoff, it should also stay in the air. At least there will be some form of control for the thrust, since just fixing the value is not going to do what you want I think.

**Reply by 7enHom** [reply]
Thanks!

I probably understood. I think this would work for me if I could have the crazyflie control its height by controlling thrust. Then If I build the firmware with [CONFIG_CONTROLLER_PID_IMPROVED_BARO_Z_HOLD] enabled in kbuild, how can I set the height and the thrust setpoint I want? And..., I don't know how to send setpoints to crazyflie using the client, or I don't know how to enable height hold when using a script instead of a client.

**Reply by 7enHom** [reply]
And by the way, for the two commands I run it just like this:
`
for y in range(30):
    cf.commander.send_zdistance_setpoint(0, 0, 0, y / 60)
    time.sleep(0.1)

 for _ in range(20):
    cf.commander.send_setpoint(0, 0, 0, 40000)
    time.sleep(0.1)
`
But it does not work continuous, it drops and then taking off again. Is there a problem here?

**Reply by knmcguire** [reply]
It's the assisted mode selection I am referring to that you can use with an handheld controller: https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/flightcontrol_tab/

But about your script, are you sure that the thrust at that point is at value 40000 at the end of the first if loop? Are you monitoring [the thrust through the logging variables](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/logs/#controllercmd_thrust)?
